clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -160,7 +161,8 @@ public NetworkClient(Selectable selector,
         this(metadataUpdater, null, selector, clientId, maxInFlightRequestsPerConnection,
              reconnectBackoffMs, reconnectBackoffMax,
              socketSendBuffer, socketReceiveBuffer, requestTimeoutMs, time,
-             discoverBrokerVersions, apiVersions, null);
+             discoverBrokerVersions, apiVersions, null,
+                new LogContext("[NetworkClient clientId=" + clientId + "] "));
```

Comment:
Maybe we can receive a `LogContext` in this constructor as well. This constructor is typically used for inter-broker communication and the callers can provide a useful context too.

Reply:
@ijuma 0847add